lib/net/wifi.toit

```diff
@@ -53,6 +101,33 @@ establish config/Map? -> Interface:
   if not service: throw "WiFi unavailable"
   return WifiInterface_ service (service.establish config)
 
+scan channels/ByteArray --passive/bool=false --period_per_channel_ms/int=SCAN_TIMEOUT_MS_ -> List:
+  if channels.size < 1: throw "Channels are unspecified"
+
+  service := service_
+  if not service: throw "WiFi unavailable"
+
+  config ::= {
+    CONFIG_SCAN_PASSIVE: passive,
+    CONFIG_SCAN_CHANNELS: channels,
+    CONFIG_SCAN_PERIOD: period_per_channel_ms,
+  }
+
+  ap_list := []
+  data_list := service.scan config
+  ap_count := data_list.size / WIFI_SCAN_ELEMENT_COUNT_
+  ap_count.repeat:
+    offset := it * WIFI_SCAN_ELEMENT_COUNT_
+    ap := AccessPoint
+        --ssid=data_list[offset + WIFI_SCAN_SSID_]
+        --bssid=data_list[offset + WIFI_SCAN_BSSID_]
+        --rssi=data_list[offset + WIFI_SCAN_RSSI_]
+        --authmode=data_list[offset + WIFI_SCAN_AUTHMODE_]
+        --channel=data_list[offset + WIFI_SCAN_CHANNEL_]
+    ap_list.add ap
+
+  return ap_list
```

Comment:
You didn't like my suggestion for allocating the resulting list with the right size and returning it?

```suggestion
  data_list := service.scan config
  ap_count := data_list.size / WIFI_SCAN_ELEMENT_COUNT_
  return List ap_count:
    offset := it * WIFI_SCAN_ELEMENT_COUNT_
    AccessPoint
        --ssid=data_list[offset + WIFI_SCAN_SSID_]
        --bssid=data_list[offset + WIFI_SCAN_BSSID_]
        --rssi=data_list[offset + WIFI_SCAN_RSSI_]
        --authmode=data_list[offset + WIFI_SCAN_AUTHMODE_]
        --channel=data_list[offset + WIFI_SCAN_CHANNEL_]
```